### PR TITLE
Allow negative prices in aggregation

### PR DESCRIPTION
**Pull request by jayantk**
We want negative prices for yields. The only blocker for negative numbers is an overly-aggressive `if` statement in the aggregation logic designed to prevent overflows. I updated this if statement to the minimum necessary conditions to prevent overflow.

Note that the aggregation algorithm (`price_model`) is already tested with negative numbers (see test_price_model.c) so this should be fine. I also added a test to make sure negative numbers get aggregated properly, and changed some of the existing test cases that include negative numbers.

There was also a temporary check in the ema calculation that prevented negative numbers. I removed that check and added reasonably extensive tests for negative emas to validate the fix.

### CODE REVIEW

File: program/c/src/oracle/upd_aggregate.h [hunk 1]
```diff
@@ -102,19 +102,12 @@ static void upd_ema(
     // compute numer/denom and new value from decay factor
     pd_load( numer, ptr->numer_ );
     pd_load( denom, ptr->denom_ );
-    if ( numer->v_ < 0 || denom->v_ < 0 ) {
```
Comment (jayantk) [inline program/c/src/oracle/upd_aggregate.h:105]: this code was a temporary thing when we changed up the ema early on. doesn't need to exist now, and also prevents negative ema values from working properly.

File: program/rust/test_data/aggregation/2.result [hunk 2]
```diff
@@ -1 +1 @@
-{"exponent":-3,"price":0,"conf":0,"status":"unknown"}
```
Comment (jayantk) [inline program/rust/test_data/aggregation/2.result:1]: all of these changed tests have input prices that are either 0 or negative

File: program/rust/src/lib.rs [hunk 3]
```diff
@@ -2,6 +2,8 @@
 // Allow non upper case globals from C
 #![allow(non_upper_case_globals)]
 
+extern crate core;
```
Comment (guibescos) [inline program/rust/src/lib.rs:5]: I don't get this

**Review by guibescos — DISMISSED** [review]
